go/api/response_types.go

```diff
@@ -146,3 +151,61 @@ func CreateEmptyClusterValue() ClusterValue[interface{}] {
 		value: Result[interface{}]{val: empty, isNil: true},
 	}
 }
+
+type XPendingSummary struct {
+	NumOfMessages    int64
+	StartId          Result[string]
+	EndId            Result[string]
+	ConsumerMessages Result[[]ConsumerPendingMessage]
+}
+
+type ConsumerPendingMessage struct {
+	ConsumerName string
+	MessageCount int64
+}
+
+type XPendingDetail struct {
+	Id            string
+	ConsumerName  string
+	IdleTime      int64
+	DeliveryCount int64
+}
+
+func CreateXPendingSummary() *XPendingSummary {
+	return &XPendingSummary{0, CreateNilStringResult(), CreateNilStringResult(), CreateNilConsumerPendingMessagesResult()}
+}
+
+func CreateNilXPendingSummary() XPendingSummary {
+	return XPendingSummary{0, CreateNilStringResult(), CreateNilStringResult(), CreateNilConsumerPendingMessagesResult()}
+}
+
+func CreateConsumerPendingMessagesResult(pendingMessages []interface{}) Result[[]ConsumerPendingMessage] {
```

Comment:
I think this code should be in response handlers

Reply:
Done